Missing Boost::System on ARM Ubuntu 18.04

Following the Ubuntu [instructions](https://github.com/bitcoin/bitcoin/blob/master/doc/build-unix.md#ubuntu--debian) on an Orange Pi Plus 2e running Ubuntu Bionic (Armbian):

```
sudo apt-get install build-essential libtool autotools-dev automake pkg-config bsdmainutils python3 ccache
sudo apt-get install libssl-dev libevent-dev libboost-system-dev libboost-filesystem-dev libboost-chrono-dev libboost-test-dev libboost-thread-dev
./autogen.sh 
./configure --disable-wallet --without-gui --disable-tests --disable-bench
...
checking for boostlib >= 1.47.0 (104700)... yes
checking whether the Boost::System library is available... yes
configure: error: Could not find a version of the Boost::System library!
```

```
Linux orangepiplus2e 4.19.57-sunxi #5.90 SMP Fri Jul 5 17:58:43 CEST 2019 armv7l armv7l armv7l GNU/Linux
```

As suggested on [Stack Overflow](https://stackoverflow.com/a/44691993), adding `--with-boost-libdir=/usr/lib/arm-linux-gnueabihf` works around the issue. But ideally `./configure` should figure this out.

I've installed Bitcoin Core on the same type of device before, also using Armbian, though this was over a year ago. I did not need the workaround back then.

--- a/build-aux/m4/ax_boost_base.m4
+++ b/build-aux/m4/ax_boost_base.m4
@@ -33,7 +33,7 @@
 #   and this notice are preserved. This file is offered as-is, without any
 #   warranty.
 
-#serial 47
+#serial 48
 
 # example boost program (need to pass version)
 m4_define([_AX_BOOST_BASE_PROGRAM],
@@ -123,6 +123,7 @@ AC_DEFUN([_AX_BOOST_BASE_RUNDETECT],[
     dnl are almost assuredly the ones desired.
     AS_CASE([${host_cpu}],
       [i?86],[multiarch_libsubdir="lib/i386-${host_os}"],
+      [armv7l],[multiarch_libsubdir="lib/arm-${host_os}"],
       [multiarch_libsubdir="lib/${host_cpu}-${host_os}"]
     )
 

